bokehjs/src/coffee/node_main.coffee

```diff
@@ -0,0 +1,43 @@
+path = require "path"
+assert = require "assert"
+rootRequire = require("root-require")
+
+root = rootRequire.packpath.parent()
+pkg = rootRequire("./package.json")
+
+module.constructor.prototype.require = (modulePath) ->
+  assert(modulePath, 'missing path')
+  assert(typeof modulePath == 'string', 'path must be a string')
+
+  load = (modulePath) =>
+    this.constructor._load(modulePath, this)
+
+  overridePath = pkg.browser[modulePath]
+
+  if overridePath?
+    modulePath = path.join(root, overridePath)
+
+  return load(modulePath)
+
+if not (global.window? and global.document?)
+  jsdom = require('jsdom').jsdom
```

Comment:
So, it looks like the only problem I can see right now, is that there there is one `main` field in `package.json` , but there are two conflicting entry points for either node.js and browserify. If webpack can be told to override the default `main`, then there should be no problem at all.